org.opentosca.bus.management.api.osgievent/src/org/opentosca/bus/management/api/osgievent/MBEventHandler.java

```diff
@@ -157,8 +222,518 @@ public void handleEvent(final Event event) {
+                                                   Collection<String> sourceRelationIDs,
+                                                   Collection<String> targetNodeIDs,
+                                                   Collection<String> targetRelationIDs) {
+        ICoreEndpointService endpointService = this.getEndpointService();
+        for (WSDLEndpoint endpoint : endpointService.getWSDLEndpoints()) {
+            Collection<String> sourceNodesMetadata = toStringCollection(endpoint.getMetadata().get("SOURCENODES"), ",");
+            Collection<String> sourceRelationsMetadata =
+                toStringCollection(endpoint.getMetadata().get("SOURCERELATIONS"), ",");
+            Collection<String> targetNodesMetadata = toStringCollection(endpoint.getMetadata().get("TARGETNODES"), ",");
+            Collection<String> targetRelationsMetadata =
+                toStringCollection(endpoint.getMetadata().get("TARGETRELATIONS"), ",");
+
+            if (sourceNodeIDs.equals(sourceNodesMetadata) && sourceRelationIDs.equals(sourceRelationsMetadata)
+                && targetNodeIDs.equals(targetNodesMetadata) && targetRelationIDs.equals(targetRelationsMetadata)) {
+                return endpoint;
+            }
+        }
+
+        return null;
+    }
+
+    private Map<String, String> toEndpointMetadata(Collection<String> sourceNodeIDs,
+                                                   Collection<String> sourceRelationIDs,
+                                                   Collection<String> targetNodeIDs,
+                                                   Collection<String> targetRelationIDs) {
+        Map<String, String> result = new HashMap<String, String>();
+
+        result.put("SOURCENODES", this.toCSV(sourceNodeIDs));
+        result.put("SOURCERELATIONS", this.toCSV(sourceRelationIDs));
+        result.put("TARGETNODES", this.toCSV(targetNodeIDs));
+        result.put("TARGETRELATIONS", this.toCSV(targetRelationIDs));
+
+        return result;
+    }
+
+    private Collection<String> toStringCollection(String data, String separator) {
+        Collection<String> result = new ArrayList<String>();
+
+        if(data == null || data.isEmpty()) {
+            return result;
+        }
+        
+        String[] split = data.split(separator);
+
+        for (String part : split) {
+            if (part != null && !part.equals("") && !part.isEmpty()) {
+                result.add(part);
+            }
+        }
+
+        return result;
+    }
+
+    private String toCSV(Collection<String> strings) {
+        StringBuilder strB = new StringBuilder();
+
+        strings.forEach(s -> strB.append(s).append(","));
+
+        return strB.toString();
+    }
+
+    private Set<PlanInstanceInput> toPlanInstanceInputs(Map<String, String> inputs) {
+        Set<PlanInstanceInput> result = new HashSet<PlanInstanceInput>();
+
+        for (String key : inputs.keySet()) {
+            String val = inputs.get(key);
```

Comment:
it's cleaner to iterate over the entry set here. Alternatively you could even use `forEach`:

`inputs.forEach((key, value) -> result.add(new PlanInstanceInput(key, value, "string")));`

Reply:
Changed (Although not a huge fan of mixing code paradigms ;) )